### PR TITLE
Add recaptcha to forms

### PR DESCRIPTION
**Pull request by atomtay**
Add invisible reCAPTCHA v2 to login, password reset, and registration

## Description
Added the reCAPTCHA gem and added it to the login, password reset, and sign up forms. Added relevant controllers to override those from Devise.

## Reviewer Note: Did I run `rails test` and 100% passing tests before merging?
Yes, and 2 tests are currently failing in rspec. However, it appears to be an issue in factories that is being examined in #205 .

### CODE REVIEW

File: app/models/school.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 require 'json'
-require 'rgeo/geos/geos_c_impl'
+# require 'rgeo/geos/geos_c_impl'
```
Comment (mzagaja) [inline app/models/school.rb]: I think this change creeped back into the PR, we should go back to requiring this if I remember right.

File: db/migrate/20170125015659_create_districts.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ class CreateDistricts < ActiveRecord::Migration
 
   def change
     enable_extension "postgis"
-    enable_extension "pgrouting"
+    # enable_extension "pgrouting"
```
Comment (mzagaja) [inline db/migrate/20170125015659_create_districts.rb]: I also thought your force push removed the commenting out of `pg_routing` so should check this as well.

Comment (atomtay) [reply db/migrate/20170125015659_create_districts.rb]: weird, I had thought so too. Maybe that was a different branch and I didn't notice? In any case, I fixed that in the newest commit.

File: app/models/school.rb (hunk 1 above)
Comment (atomtay) [reply app/models/school.rb]: Strange! Went back and fixed that up.

**Comment by atomtay** [comment]
Updated the readme with instructions to get the reCAPTCHA keys and how to resolve the proj brew error on MacOS.

**Pull request merged**